### ISSUE TITLE
Add an OAuth example to the new SvelteKit SSR documentation

### ISSUE THREAD

**Issue by coppinger** [supabase/supabase#23618]
# Improve documentation

https://supabase.com/docs/guides/auth/server-side/sveltekit

https://github.com/supabase/supabase/blob/master/apps/docs/content/guides/auth/server-side/sveltekit.mdx

## Describe the problem

I found it slightly difficult to figure out how to use OAuth with the new SvelteKit SSR guide as there isn't an example of how to handle it—seemingly by redirecting to data.url

ex:
```
		if (error) {
			console.error(error);
			return redirect(303, '/auth/error');
		} else {
			return redirect(303, data.url);
		}
```

## Describe the improvement

It'd be great to address this in the guide. I tried to make a PR, but couldn't quite figure out how best to approach adding this info.

**Comment by Aishwary2004Gupta** [comment]
Hey, can I work on this issue?

**Comment by charislam** [comment]
@Aishwary2004Gupta, of course! We don't assign issues, so feel free to open a PR if no one else has gotten there first! 😄

**Comment by ShubhSingh07** [comment]
Hi, I'd like to work on this issue. Could you please assign it to me?

**Comment by melaksenay** [comment]
> Hi, I'd like to work on this issue. Could you please assign it to me?

For future reference, there is no assigning people to issues! You can work on them as you please and just make sure to let others know you are working on a given issue by commenting!

**Comment by sarthaksharma27** [comment]
Chat! I am working on this issue #23618

**Comment by Khan4218** [comment]
Hi! I’d like to work on this issue. Can I get assigned?

**Comment by nikhil008-git** [comment]
Heyy, can i work on this issue?

**Comment by dhunganapramod9** [comment]
Hey! i am working on this

**Comment by aparajitaks** [comment]
Hi! I’d like to work on this issue.

I can add an OAuth example for the new SvelteKit SSR documentation. I’ll prepare a clear, minimal example and open a PR once it’s ready.

Please let me know if there are any specific requirements or preferred formats for documentation examples.

**Comment by RudraKo** [comment]
Hello! I’d like to take this issue.

I can implement the OAuth example for the updated SvelteKit SSR docs. I’ll follow the existing documentation structure and open a PR soon.

If there are any guidelines I should follow for doc contributions, I’d be happy to review them.

**Comment by harikapadia999** [comment]
Hi! I'd like to work on this documentation issue.

**Plan:**
I'll add a comprehensive OAuth example to the SvelteKit SSR documentation covering:
1. OAuth provider setup (Google, GitHub, etc.)
2. Server-side OAuth flow implementation
3. Callback handling in SvelteKit
4. Session management with cookies
5. Protected routes example

I'll follow the existing documentation structure and ensure the example is production-ready. Will submit a PR shortly!

**Comment by tysoncung** [comment]
Documentation improvements are always valuable! I'd be happy to help with this. What specific areas do you think need better documentation?

**Comment by FabioArieiraBaia** [comment]
Hello! 👋

I've added a comprehensive OAuth example to the SvelteKit SSR documentation as requested in this issue. Created a PR with the solution:

**What was added:**
- ✅ OAuth callback route implementation example
- ✅ Server action example for initiating OAuth sign-in
- ✅ Client component example for OAuth sign-in
- ✅ Clear explanation of the OAuth flow with step-by-step process

The example shows how to properly handle redirects using SvelteKit's \
edirect()\ function and demonstrates both server-side and client-side OAuth initiation patterns.

The PR is ready for review. Any feedback is welcome!

---

**Developed by:** Fábio Arieira  
**Website:** https://fabioarieira.com  
**Production Projects:**
- IA-Books: https://iabooks.com.br
- ViralFlow: https://fabioarieira.com/viralflow
- Real Estate Platform: https://fabioarieira.com/imob

**Comment by nidhisoni01** [comment]
Hi 👋  

I’d like to work on improving the SvelteKit SSR authentication documentation.

**Problem**
While reviewing the SvelteKit SSR auth guide, I noticed that there is no example demonstrating how to implement OAuth authentication with proper server-side redirect handling. This makes it difficult for developers to understand the correct SSR flow for OAuth.

**Steps to reproduce**
1. Visit the Supabase SvelteKit SSR authentication guide in the docs.
2. Review the authentication examples provided.
3. Observe that there is no OAuth-specific example covering redirect handling in an SSR context.
4. Attempt to implement OAuth in a SvelteKit SSR app and encounter confusion due to the lack of a reference example.

**Proposed improvement**
I can add a clear OAuth example that:
- Demonstrates initiating OAuth in an SSR-safe way
- Handles redirects correctly in SvelteKit
- Includes brief explanations for each step

Please assign this issue to me — I’d be happy to submit a PR.
Thanks! 😊

**Comment by aasthachauhan004** [comment]
Is this issue resolved? As i can see there are multiple PRs for the same.

**Comment by RohanHandore** [comment]
Hi Chat ..has this issue been resolved? I’d like to start working on it. If someone’s already working on it, please let me know who so I don’t duplicate effort.

**Comment by ChrisChinchilla** [comment]
Thanks to everyone who submitted a PR for this. As you might be able to see, the page mentioned changed quite a bit in the past few months, and right now, we are trying to follow a consistent example for all of the SDKs in the SSR guide. For now, we're keeping it simple and are considering how to add to that simple example in the future. I also wonder if OAuth examples may be better in the OAuth section as opposed to in the SSR guide?

For example: https://supabase.com/docs/guides/auth/social-login/auth-github?queryGroups=framework&framework=sveltekit&queryGroups=environment&environment=server

 I will close this issue for now, as there are a lot of duplicate PRs to address. https://github.com/supabase/supabase/pull/42168 and https://github.com/supabase/supabase/pull/42017/changes follow the current page structure the most, so I will probably keep those open and use elements of them.

Thanks!
